repl-web/repl.lisp

```diff
@@ -29,42 +29,40 @@
 
 ;;; Decides wheater the input the user has entered is completed or we
 ;;; should accept one more line.
-(defun indent-level (string)
-  (let ((i 0)
-        (stringp nil)
-        (s (length string))
-        (depth 0))
-
-    (while (< i s)
-      (cond
-        (stringp
-         (case (char string i)
-           (#\\
+(defun %sexpr-complete (string)
+    (let ((i 0)
+          (stringp nil)
+          (comments nil)
+          (s (length string))
+          (depth 0))
+        (while (< i s)
+            (cond
+              (comments 
+               (case (char string i)
+                 (#\newline 
+                  (setq comments nil))))
+              (stringp
+               (case (char string i)
+                 (#\\
+                  (incf i))
+                 (#\"
+                  (setq stringp nil)
+                  (decf depth))))
+              (t
+               (case (char string i)
+                 (#\; 
+                  (setq comments t))
+                 (#\( (unless comments (incf depth)))
+                 (#\) (unless comments (decf depth)))
+                 (#\"
+                  (incf depth)
+                  (setq stringp t)))))
             (incf i))
-           (#\"
-            (setq stringp nil)
-            (decf depth))))
-        (t
-         (case (char string i)
-           (#\( (incf depth))
-           (#\) (decf depth))
-           (#\"
-            (incf depth)
-            (setq stringp t)))))
-      (incf i))
-
-    (if (and (zerop depth))
-        nil
-        ;; We should use something based on DEPTH in order to make
-        ;; edition nice, but the behaviour is a bit weird with
-        ;; jqconsole.
-        0)))
-
-
+        (if (<= depth 0)
+            nil
+            0)))
 
 (defun toplevel ()
-  (#j:jqconsole:RegisterMatching "(" ")" "parents")
```

Comment:
Why did we delete the matching parenthesis?  I think that was a nice functionality.